src/lib/input/input.md

```diff
@@ -33,107 +30,36 @@ be used with `matInput`:
 
-A placeholder is an indicative text displayed in the input zone when the input does not contain
-text. When text is present, the indicative text will float above this input zone.
-
-The `floatPlaceholder` attribute of `mat-form-field` can be set to `never` to hide the
-indicative text instead when text is present in the input.
-
-When setting `floatPlaceholder` to `always`, the floating label will always show above the input.
-
-A placeholder for the input can be specified in one of two ways: either using the `placeholder`
-attribute on the `input` or `textarea`, or using an `mat-placeholder` element in the
-`mat-form-field`. Using both will raise an error.
-
-Global default placeholder options can be specified by setting the `MAT_PLACEHOLDER_GLOBAL_OPTIONS`
-provider. This setting will apply to all components that support the floating placeholder.
-
-```ts
-@NgModule({
-  providers: [
-    {provide: MAT_PLACEHOLDER_GLOBAL_OPTIONS, useValue: {float: 'always'}}
-  ]
-})
-```
-
-Here are the available global options:
-
-| Name            | Type    | Values              | Description                               |
-| --------------- | ------- | ------------------- | ----------------------------------------- |
-| float           | string  | auto, always, never | The default placeholder float behavior.   |
-
-### Prefix and Suffix
-
-HTML can be included before and after the input tag, as a prefix or suffix. It will be underlined
-as per the Material specification, and clicking it will focus the input.
+A placeholder is a text label displayed in the input area when the input does not contain text.
+When text is present, the placeholder will float above the input area. The placeholder can be
+specified either via a `placeholder` attribute on the input or a `<mat-placeholder>` element in the
+same form field as the `matInput`. The `<mat-form-field>` also has additional options for changing
+the behavior of the placeholder. For more information see the
+[form field placeholder documentation](https://material.angular.io/components/form-field/overview#floating-placeholder).
 
-Adding the `matPrefix` attribute to an element inside the `mat-form-field` will designate it as
-the prefix. Similarly, adding `matSuffix` will designate it as the suffix.
+### Changing when error messages are shown
 
-<!-- example(input-prefix-suffix) -->
+The `<mat-form-field>` allows you to
+[associate error messages](https://material.angular.io/components/form-field/overview#error-messages)
+with your `matInput`. By default, these error messages are shown when the control is invalid and
+either the user has interacted with (touched) the element or the parent form has been submitted. If
+you wish to override this behavior (e.g. to show the error as soon as the invalid control is dirty
+or when a parent form group is invalid), you can use the `errorStateMatcher` property of the
+`matInput`. The property takes an instance of an `ErrorStateMatcher` object. An `ErrorStateMatcher`
```

Comment:
I think the more common use-case for customizing the `ErrorStateMatcher` is overriding the global one, rather than doing it per-control. Maybe we should have an example for that case as well?

Reply:
Unfortunately our examples aren't that flexible atm. We don't get to mess with the module. That's why I left an inline code snippet showing how to do it globally.